Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -1274,65 +553,252 @@ extension RawSyntax {
+  ///   - text: Token text.
+  ///   - leadingTrivia: Leading trivia.
+  ///   - trailingTrivia: Trailing trivia.
+  static func makeMaterializedToken(
+    arena: SyntaxArena,
+    kind: TokenKind,
+    leadingTrivia: Trivia,
+    trailingTrivia: Trivia
+  ) -> RawSyntax {
+    let decomposed = kind.decomposeToRaw()
+    let rawKind = decomposed.rawKind
+    let text: SyntaxText = (decomposed.string.map({arena.intern($0)}) ??
+                            decomposed.rawKind.defaultText ??
+                            "")
+
+    var byteLength = text.count
+
+    let triviaPieces = makeRawTriviaPieces(
+      arena: arena, leadingTrivia: leadingTrivia, trailingTrivia: trailingTrivia)
+
+    byteLength += triviaPieces.byteLength
+
+    return .materializedToken(
+      arena: arena, kind: rawKind, text: text,
+      triviaPieces: triviaPieces.pieces,
+      numLeadingTrivia: numericCast(leadingTrivia.count),
+      byteLength: numericCast(byteLength))
+  }
+
+  static func makeEmptyToken(
+    arena: SyntaxArena,
+    kind: TokenKind
+  ) -> RawSyntax {
+    let (rawKind, _) = kind.decomposeToRaw()
+    return .materializedToken(
+      arena: arena, kind: rawKind, text: "",
+      triviaPieces: .init(start: nil, count: 0),
+      numLeadingTrivia: 0,
+      byteLength: 0)
+  }
+}
+
+extension RawSyntax {
+  /// "Designated" factory method to create a layout node.
+  ///
+  /// This should not be called directly.
+  /// Use `makeLayout(arena:kind:uninitializedCount:initializingWith:)` or
+  /// `makeEmptyLayout(arena:kind:)` instead.
+  ///
+  /// - Parameters:
+  ///   - arena: SyntaxArea to the result node data resides.
+  ///   - kind: Syntax kind. This should not be `.token`.
+  ///   - layout: Layout buffer of the children.
+  ///   - byteLength: Computed total byte length of this node.
+  ///   - descedantCount: Total number of the descendant nodes in `layout`.
+  fileprivate static func layout(
+    arena: SyntaxArena,
+    kind: SyntaxKind,
+    layout: RawSyntaxBuffer,
+    byteLength: Int,
+    descendantCount: Int
+  ) -> RawSyntax {
+    let payload = RawSyntaxData.Layout(
+      kind: kind, layout: layout,
+      byteLength: byteLength, descendantCount: descendantCount)
+    return RawSyntax(arena: arena, payload: .layout(payload))
+  }
+
+  /// Factory method to create a layout node.
+  ///
+  /// - Parameters:
+  ///   - arena: SyntaxArea to the result node data resides.
+  ///   - kind: Syntax kind.
+  ///   - count: Number of children.
+  ///   - initializer: A closure that initializes elements.
+  static func makeLayout(
+    arena: SyntaxArena,
+    kind: SyntaxKind,
+    uninitializedCount count: Int,
+    initializingWith initializer: (UnsafeMutableBufferPointer<RawSyntax?>) -> Void
+  ) -> RawSyntax {
+    // Allocate and initialize the list.
+    let layoutBuffer = arena.allocateRawSyntaxBuffer(count: count)
+    initializer(layoutBuffer)
+    // validateLayout(layout: RawSyntaxBuffer(layoutBuffer), as: kind)
```

Comment:
Do you intend to add the validation again or are we dropping it?

Reply:
Adding. To be precise, I'm moving the validation from `Syntax` to `RawSyntax`.
https://github.com/rintaro/swift-syntax/blob/bump-ptr-allocation-and-everything/Sources/SwiftSyntax/gyb_generated/SyntaxValidation.swift 
And do it whenever it's created.

Reply:
OK, I’m fine with adding validation again in a follow-up PR.